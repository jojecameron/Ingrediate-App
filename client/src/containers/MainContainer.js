import React, { Component } from 'react';
import IngredientForm from '../components/IngredientForm';
import RecipeContainer from './RecipeContainer';
import DishForm from '../components/DishForm';
import Loading from '../components/Loading';

const url = 'http://localhost:3000/generate'

class MainContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.state.ingredients = [];
    this.state.dishType = [];
    this.state.recipeList = [];
    this.state.isLoading = false;
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleDishChange = this.handleDishChange.bind(this);
    this.deleteRecipe = this.deleteRecipe.bind(this);
    this.favoriteRecipe = this.favoriteRecipe.bind(this);
    this.sendIngredientsToServer = this.sendIngredientsToServer.bind(this)
  }

  // add logic to delete recipe from display
  deleteRecipe(recipe) {
    console.log('deleteRecipe');
  }
  
  // add logic to favorite a recipe on display
  favoriteRecipe(recipe) {
    console.log('favoriteRecipe');
  }

  // preps dishType and ingredients to be sent to server
  handleSubmit(event) {
    event.preventDefault();
    // console.log('handleSubmit is running');
    const listOfIngredients = [];
    this.state.ingredients.forEach(ingredient => {
        listOfIngredients.push(ingredient.label);
    })
    const { dishType } = this.state;
    const dishAndIngredients = dishType.concat(listOfIngredients)
    // console.log(dishAndIngredients)
    // console.log(this.state);
    // send list of ingredients to Server
    this.sendIngredientsToServer(listOfIngredients);
  }

  // updates state for ingredients
  handleChange(value) {
    // console.log('Before', this.state);
    // console.log(value)
    const newState = this.state;
    newState.ingredients = value;
    this.setState({...newState});
    // console.log('After', this.state);
  }

  // udpates state for dishType
  handleDishChange(value) {
    console.log('Before', this.state);
    const newState = this.state;
    newState.dishType = [ value ];
    this.setState({...newState});
    console.log('After', this.state);
  }

  // makes post request to server, handles loading state change, receives data and udpates state
  sendIngredientsToServer = async (ingredients) => {
    if (!ingredients.length) {
      return alert('Please enter ingredients...');
    }
    console.log('ingredients to send to Server..', ingredients);
    const newState = this.state;
    newState.isLoading = true;
    this.setState({...newState});
    try {
      const result = await fetch(url, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        // mode: 'no-cors',
        body: JSON.stringify( ingredients )
      })
      const data = await result.json();
      console.log(data);
      newState.recipeList.push(data);
      newState.isLoading = false;
      // console.log(newState.recipes);
      this.setState({...newState});
      console.log(this.state.recipeList);
    } catch (err) {
      console.log(err);
    }
  }

  render() {
    return (
      <div className="MainContainer">
        <h1><em>Ingrediate</em><br/><span>Recipe Generator</span></h1>
        <DishForm 
          id = 'DishForm'
          handleDishChange = {this.handleDishChange}
        />
        <IngredientForm 
          id = 'IngredientForm'
          handleChange = {this.handleChange}
          handleSubmit = {this.handleSubmit}
        />
        {this.state.isLoading ? <Loading /> : <br/>}
        <RecipeContainer 
          recipeList = {this.state.recipeList}
          deleteRecipe = {this.deleteRecipe}
          favoriteRecipe = {this.favoriteRecipe}
        />
      </div>
    );
  }
}

export default MainContainer;
