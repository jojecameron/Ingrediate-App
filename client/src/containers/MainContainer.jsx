import React, { useState } from 'react';
import IngredientForm from '../components/IngredientForm';
import RecipeContainer from './RecipeContainer';
import DishForm from '../components/DishForm';
import Loading from '../components/Loading';

const url = 'http://localhost:3000/generate';
const favoritesUrl = 'http://localhost:3000/favorites';

const MainContainer = (props) => {
    const [ingredients, setIngredients] = useState([]);
    const [dishType, setDishType] = useState('Breakfast');
    const [recipeList, setRecipeList] = useState([]);
    const [favoriteRecipes, setFavoriteRecipes] = useState([]);
    const [isLoading, setIsLoading] = useState(false);

  // on select favorite add to favoriteRecipes array, deselect favorite removes from array
  const favoriteRecipe = (recipe) => {
    const { isFavorite, recipeIndex } = recipe;
    if (isFavorite) {
      const favoritedRecipe = recipeList[recipeIndex];
      favoriteRecipes.push({ index: recipeIndex, recipe: favoritedRecipe });
      setFavoriteRecipes([{ index: recipeIndex, recipe: favoritedRecipe }, ...favoriteRecipes]);
      // TBD should throttle call to server
      // saveFavorites({ index: recipeIndex, recipe: favoritedRecipe });
    }
    if (favoriteRecipes.length) {
      if (!isFavorite) {
        const index = favoriteRecipes.findIndex((recipe) => {
          recipe.index === recipeIndex;
        });
        setFavoriteRecipes(favoriteRecipes.splice(index, 1));
        // TBD should throttle call to server
        // this.deleteFavorite(recipeIndex);
      }
    }
  }

  const deleteRecipe = () => {
    console.log('deleted recipe');
  }

  const saveFavorites = async (favoriteObject) => {
    console.log('To send to server', favoriteObject);
    try {
      const result = await fetch(favoritesUrl, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(favoriteObject),
      });
      const data = await result.json();
      console.log('ADDED TO DATABASE: ', data);
    } catch (err) {
      console.log(err);
    }
  };

  const deleteFavorite = async (index) => {
    console.log('To send to server', index);
    try {
      const result = await fetch(favoritesUrl, {
        method: 'DELETE',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ index: index }),
      });
      const data = await result.json();
      console.log('DELETED FROM DATABASE: ', data);
    } catch (err) {
      console.log(err);
    }
  };

  // preps dishType and ingredients to be sent to server
  const handleSubmit = (event) => {
    event.preventDefault();
    const listOfIngredients = [];
    ingredients.forEach((ingredient) => {
      listOfIngredients.push(ingredient.label);
    });
    const dishAndIngredients = dishType.concat(listOfIngredients);
    console.log('this is dishAndIngredients', dishAndIngredients);
    console.log('this is listOfIngredients', listOfIngredients);
    sendIngredientsToServer(listOfIngredients);
  }

  // updates state for ingredients
  const handleChange = (value) => {
    setIngredients([value, ...ingredients ]);
  }

  // udpates state for dishType
  const handleDishChange = (value) => {
    setDishType(value);
  }

  // makes post request to server, handles loading state change, receives data and udpates state
  const sendIngredientsToServer = async (ingredients) => {
    if (!ingredients.length) {
      return alert('Please enter ingredients...');
    }
    setIsLoading(true);
    try {
      const result = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(ingredients),
      });
      const data = await result.json();
      setRecipeList([data, ...recipeList]);
      setIsLoading(false);
    } catch (err) {
      console.log(err);
    }
  };

    return (
      <div className="MainContainer">
        <h1>
          <em>Ingrediate</em>
          <br />
          <span>Recipe Generator</span>
        </h1>
        <DishForm id="DishForm" handleDishChange={handleDishChange} />
        <IngredientForm
          id="IngredientForm"
          handleChange={handleChange}
          handleSubmit={handleSubmit}
          isLoading={isLoading}
        />
        {isLoading ? <Loading /> : <br />}
        <RecipeContainer
          recipeList={recipeList}
          deleteRecipe={deleteRecipe}
          favoriteRecipe={favoriteRecipe}
        />
      </div>
    );
};

export default MainContainer;
