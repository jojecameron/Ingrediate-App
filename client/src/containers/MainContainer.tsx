import { useState } from 'react';
import RecipeContainer from './RecipeContainer';
import IngredientForm from '../components/IngredientForm';
import DishForm from '../components/DishForm';
import Loading from '../components/Loading';
import Header from '../components/Header';
import Modal from '../components/Modal';
import { DishType, Recipe, Favorite } from '../types';

const url = 'http://localhost:3000/generate';
const favoritesUrl = 'http://localhost:3000/favorites';

interface Ingredient {
  label: string;
}

const MainContainer = (): JSX.Element => {
  const [ingredientChoices, setIngredientChoices] = useState<Ingredient[]>([]);
  const [dishType, setDishType] = useState<DishType>('breakfast');
  const [recipeList, setRecipeList] = useState<Recipe[]>([]);
  const [favoriteRecipes, setFavoriteRecipes] = useState<Favorite[]>([]);
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const [openModal, setOpenModal] = useState<boolean>(false);
  const [modalState, setModalState] = useState<string>('Log in');

  const favoriteRecipe = (isFavorite: boolean, recipe: Favorite) => {
    setFavoriteRecipes((currentFavorites) => {
      if (isFavorite) {
        return [...currentFavorites, recipe];
      } else {
        return currentFavorites.filter(
          (fav) => fav.recipeTitle !== recipe.recipeTitle,
        );
      }
    });
  };

  // stores favorited recipes in db
  const saveFavorites = async (favorites: Favorite[]) => {
    try {
      const result = await fetch(favoritesUrl, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(favoriteRecipes),
      });
      const data = await result.json();
      console.log('ADDED TO DATABASE: ', data);
    } catch (err) {
      console.log(err);
    }
  };

  // preps dishType and ingredients to be sent to server
  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const listOfIngredients = [];
    ingredientChoices.forEach((ingredient) => {
      listOfIngredients.push(ingredient.label);
    });
    listOfIngredients.push(dishType);
    sendIngredientsToServer(listOfIngredients);
  };

  // makes post request to server, handles loading state change, receives data and udpates state
  const sendIngredientsToServer = async (ingredients: string[]) => {
    if (!ingredients.length) {
      return alert('Please enter ingredients...');
    }
    setIsLoading(true);
    try {
      const result = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(ingredients),
      });
      const data = await result.json();
      setRecipeList([data, ...recipeList]);
      setIsLoading(false);
    } catch (err) {
      console.log(err);
    }
  };

  const deleteRecipe = () => {
    console.log('deleted recipe');
  };

  return (
    <>
      <Header 
      setOpenModal={setOpenModal}
      setModalState={setModalState}
      />
      <section className="MainContainer">
        {!openModal ? (
          <></>
        ) : (
          <>
            <Modal 
            setOpenModal={setOpenModal}
            modalState={modalState} />
          </>
        )}
        <h1>
          <em>Ingrediate</em>
          <br />
          <span>Recipe Generator</span>
        </h1>
        <DishForm setDishType={setDishType} dishType={dishType} />
        <IngredientForm
          handleSubmit={handleSubmit}
          ingredientChoices={ingredientChoices}
          setIngredientChoices={setIngredientChoices}
        />
        {isLoading ? <Loading /> : <br />}
        <RecipeContainer
          recipeList={recipeList}
          deleteRecipe={deleteRecipe}
          favoriteRecipe={favoriteRecipe}
        />
      </section>
    </>
  );
};

export default MainContainer;
