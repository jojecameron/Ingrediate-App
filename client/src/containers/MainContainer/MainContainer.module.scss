.MainContainer {
  align-items: center;
  background-color: $primaryGreen;
  border-radius: px;
  display: flex;
  flex-direction: column;
  height: fit-content;
  justify-content: flex-start;
  min-height: 100vh;
  width: 100vw;
}

.overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 999;
}
