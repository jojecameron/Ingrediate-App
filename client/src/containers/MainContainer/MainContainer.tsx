import { useState } from 'react';
import {
  RecipeContainer,
  IngredientForm,
  DishForm,
  Loading,
  Header,
  Modal,
} from '../../components';
import {
  DishType,
  Recipe,
  Favorite,
  Ingredient,
  ModalState,
  User,
} from '../../types';
import { generateRecipe } from '../../utils/apiUtils';

const MainContainer = (): JSX.Element => {
  const [ingredientChoices, setIngredientChoices] = useState<Ingredient[]>([]);
  const [dishType, setDishType] = useState<DishType>('breakfast');
  const [recipeList, setRecipeList] = useState<Recipe[]>([]);
  const [favoriteRecipes, setFavoriteRecipes] = useState<Favorite[]>([]);
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const [modalState, setModalState] = useState<ModalState>({
    isOpen: false,
    modalType: 'Log in',
  });
  const [isLoggedIn, setIsLoggedIn] = useState<User>({
    loggedIn: false,
    display_name: '',
    email: '',
    user_id: '',
    firebase_uid: '',
  });

  // adds or removes favorited recipes from state
  const favoriteRecipe = (isFavorite: boolean, recipe: Favorite) => {
    deleteRecipe(recipe.id);
    setFavoriteRecipes((currentFavorites) => {
      if (isFavorite) {
        return [...currentFavorites, recipe];
      } else {
        return currentFavorites.filter(
          (fav) => fav.recipeTitle !== recipe.recipeTitle,
        );
      }
    });
  };

  // stores favorited recipes in db
  const saveFavorites = async () => {
    const favoritesUrl = 'http://localhost:3000/favorites';
    if (isLoggedIn.loggedIn) {
      try {
        const result = await fetch(favoritesUrl, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            favorites: favoriteRecipes,
            user_id: isLoggedIn.user_id,
          }),
        });
      } catch (err) {
        console.error(err);
      }
    } else {
      alert('Please log in to save favorites');
    }
  };

  // preps dishType and ingredients to be sent to server
  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const listOfIngredients = [];
    ingredientChoices.forEach((ingredient) => {
      listOfIngredients.push(ingredient.label);
    });
    listOfIngredients.push(dishType);
    sendIngredientsToServer(listOfIngredients);
  };

  // sends ingredients to server
  const sendIngredientsToServer = async (ingredients: string[]) => {
    if (!ingredients.length) {
      return alert('Please enter ingredients...');
    }
    setIsLoading(true);
    try {
      const result = await generateRecipe(ingredients);
      setRecipeList([result, ...recipeList]);
      setIsLoading(false);
    } catch (err) {
      console.error(err);
    }
  };

  // deletes recipe from state and favorite from db
  const deleteRecipe = async (id: string) => {
    if (favoriteRecipes.some((recipe) => recipe.id === id)) {
      try {
        const favoritesUrl = 'http://localhost:3000/favorites';
        const result = await fetch(favoritesUrl, {
          method: 'DELETE',
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            id: id,
          }),
        });
        setFavoriteRecipes((currentFavorites) => {
          return currentFavorites.filter((recipe) => recipe.id !== id);
        });
      } catch (err) {
        console.error(err);
      }
    } else {
      setRecipeList((currentRecipes) => {
        return currentRecipes.filter((recipe) => recipe.id !== id);
      });
    }
  };

  return (
    <>
      <Header
        setModalState={setModalState}
        isLoggedIn={isLoggedIn}
        setIsLoggedIn={setIsLoggedIn}
        saveFavorites={saveFavorites}
      />
      <section className="MainContainer">
        {modalState.isOpen && (
          <div
            className="overlay"
            onClick={() =>
              setModalState({ isOpen: false, modalType: 'Log in' })
            }
          />
        )}
        {!modalState.isOpen ? (
          <></>
        ) : (
          <>
            <Modal
              modalState={modalState}
              setModalState={setModalState}
              setIsLoggedIn={setIsLoggedIn}
              setFavoriteRecipes={setFavoriteRecipes}
            />
          </>
        )}
        <DishForm setDishType={setDishType} dishType={dishType} />
        <IngredientForm
          handleSubmit={handleSubmit}
          ingredientChoices={ingredientChoices}
          setIngredientChoices={setIngredientChoices}
        />
        {isLoading ? <Loading /> : <br />}
        <RecipeContainer
          recipeList={recipeList}
          deleteRecipe={deleteRecipe}
          favoriteRecipe={favoriteRecipe}
          favoriteRecipes={favoriteRecipes}
          setFavoriteRecipes={setFavoriteRecipes}
        />
      </section>
    </>
  );
};

export default MainContainer;
