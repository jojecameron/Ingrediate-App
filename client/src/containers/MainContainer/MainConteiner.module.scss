.MainContainer {
  align-items: center;
  background-color: $primaryGreen;
  border-radius: px;
  display: flex;
  flex-direction: column;
  height: fit-content;
  justify-content: flex-start;
  min-height: 100vh;
  width: 100vw;
}
