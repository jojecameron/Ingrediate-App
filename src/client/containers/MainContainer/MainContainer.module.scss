.MainContainer {
  align-items: center;
  background-color: $primaryGreen;
  border-radius: px;
  display: flex;
  flex-direction: column;
  height: fit-content;
  justify-content: flex-start;
  min-height: 100vh;
  width: 100vw;
}

.overlay {
  background-color: rgba(0, 0, 0, 0.5);
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 999;
}

.generatorConfiguration {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

@media screen and (max-width: 700px) {
  .generatorConfiguration {
    margin-bottom: 15px;
  }
}
