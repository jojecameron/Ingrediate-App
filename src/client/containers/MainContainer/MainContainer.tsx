import { useState } from 'react';
import {
  RecipeContainer,
  IngredientForm,
  DishForm,
  Loading,
  Header,
  Modal,
} from '../../components';
import {
  DishType,
  Recipe,
  Favorite,
  Ingredient,
  ModalState,
  User,
} from '../../types';
import { generateRecipe } from '../../utils/apiUtils';

const MainContainer = (): JSX.Element => {
  const [model, setModel] = useState<string>('text-davinci-003');
  const [ingredientChoices, setIngredientChoices] = useState<Ingredient[]>([]);
  const [dishType, setDishType] = useState<DishType>('breakfast');
  const [recipeList, setRecipeList] = useState<Recipe[]>([]);
  const [favoriteRecipes, setFavoriteRecipes] = useState<Favorite[]>([]);
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const [favoriteMode, setFavoriteMode] = useState<boolean>(false);
  const [modalState, setModalState] = useState<ModalState>({
    isOpen: false,
    modalType: 'Log in',
  });
  const [isLoggedIn, setIsLoggedIn] = useState<User>({
    loggedIn: false,
    displayName: '',
    email: '',
    userId: '',
    firebaseUid: '',
  });

  // adds or removes favorited recipes from state
  const favoriteRecipe = (isFavorite: boolean, recipe: Favorite) => {
    deleteRecipe(recipe.id);
    setFavoriteRecipes((currentFavorites) => {
      if (isFavorite) {
        return [...currentFavorites, recipe];
      } else {
        return currentFavorites.filter(
          (fav) => fav.recipeTitle !== recipe.recipeTitle,
        );
      }
    });
  };

  // stores favorited recipes in db
  const saveFavorites = async () => {
    const favoritesUrl = 'http://localhost:3000/favorites';
    if (isLoggedIn.loggedIn) {
      try {
        const result = await fetch(favoritesUrl, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            favorites: favoriteRecipes,
            userId: isLoggedIn.userId,
          }),
        });
      } catch (err) {
        console.error(err);
      }
    } else {
      alert('Please log in to save favorites');
    }
  };

  // preps dishType and ingredients to be sent to server
  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const listOfIngredients = [];
    ingredientChoices.forEach((ingredient) => {
      listOfIngredients.push(ingredient.label);
    });
    listOfIngredients.push(dishType);
    sendIngredientsToServer(listOfIngredients);
  };

  // sends ingredients to recipe generator service
  const sendIngredientsToServer = async (ingredients: string[]) => {
    if (!ingredients.length) {
      return alert('Please enter ingredients...');
    }
    setIsLoading(true);
    try {
      const result = await generateRecipe(ingredients, model);
      setRecipeList([result, ...recipeList]);
      setIsLoading(false);
      setFavoriteMode(false);
    } catch (err) {
      console.error(err);
    }
  };

  // deletes recipe from state and favorite from db if logged in
  const deleteRecipe = async (id: string) => {
    if (favoriteRecipes.some((recipe) => recipe.id === id)) {
      try {
        const favoritesUrl = 'http://localhost:3000/favorites';
        const result = await fetch(favoritesUrl, {
          method: 'DELETE',
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            id: id,
          }),
        });
        setFavoriteRecipes((currentFavorites) => {
          return currentFavorites.filter((recipe) => recipe.id !== id);
        });
      } catch (err) {
        console.error(err);
      }
    } else {
      setRecipeList((currentRecipes) => {
        return currentRecipes.filter((recipe) => recipe.id !== id);
      });
    }
  };

  // updates recipe title in state and db if logged in
  const updateRecipeTitle = async (
    id: string,
    newTitle: string,
    isFavorite: boolean,
  ) => {
    // for favorited recipes
    if (isFavorite) {
      // only favorites that are associated with a user
      if (isLoggedIn.loggedIn) {
        try {
          const favoritesUrl = 'http://localhost:3000/favorites';
          const result = await fetch(favoritesUrl, {
            method: 'PATCH',
            headers: {
              'Content-Type': 'application/json',
            },
            body: JSON.stringify({
              id: id,
              newTitle: newTitle,
            }),
          });
        } catch (err) {
          console.error(err);
        }
      }
      setFavoriteRecipes((currentFavorites) => {
        return currentFavorites.map((recipe) => {
          if (recipe.id === id) {
            return { ...recipe, recipeTitle: newTitle };
          }
          return recipe;
        });
      });
    } else {
      // for non-favorited recipes
      setRecipeList((currentRecipes) => {
        return currentRecipes.map((recipe) => {
          if (recipe.id === id) {
            return { ...recipe, recipeTitle: newTitle };
          }
          return recipe;
        });
      });
    }
  };

  return (
    <>
      <Header
        setModalState={setModalState}
        isLoggedIn={isLoggedIn}
        setIsLoggedIn={setIsLoggedIn}
        saveFavorites={saveFavorites}
        setFavoriteRecipes={setFavoriteRecipes}
      />
      <section className="MainContainer">
        {modalState.isOpen && (
          <div
            className="overlay"
            onClick={() =>
              setModalState({ isOpen: false, modalType: 'Log in' })
            }
          />
        )}
        {!modalState.isOpen ? (
          <></>
        ) : (
          <>
            <Modal
              modalState={modalState}
              setModalState={setModalState}
              setIsLoggedIn={setIsLoggedIn}
              setFavoriteRecipes={setFavoriteRecipes}
            />
          </>
        )}
        <DishForm setDishType={setDishType} dishType={dishType} />
        <IngredientForm
          handleSubmit={handleSubmit}
          ingredientChoices={ingredientChoices}
          setIngredientChoices={setIngredientChoices}
        />
        {isLoading ? <Loading /> : <br />}
        <RecipeContainer
          recipeList={recipeList}
          deleteRecipe={deleteRecipe}
          favoriteRecipe={favoriteRecipe}
          favoriteRecipes={favoriteRecipes}
          setFavoriteRecipes={setFavoriteRecipes}
          updateRecipeTitle={updateRecipeTitle}
          favoriteMode={favoriteMode}
          setFavoriteMode={setFavoriteMode}
        />
      </section>
    </>
  );
};

export default MainContainer;
