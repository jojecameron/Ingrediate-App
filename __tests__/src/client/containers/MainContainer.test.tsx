import React from 'react';
import { render } from '@testing-library/react'; 
import MainContainer from '../../../../src/client/containers/MainContainer/MainContainer';

test('renders a basic snapshot', () => {
  const { container } = render(<MainContainer />);
  expect(container).toMatchSnapshot(); 
});
