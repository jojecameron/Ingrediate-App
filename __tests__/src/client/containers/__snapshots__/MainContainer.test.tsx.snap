// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a basic snapshot 1`] = `
<div>
  <header>
    <div>
      <h1>
        <em>
          Ingrediate
        </em>
      </h1>
    </div>
    <div>
      <p>
        Log in
      </p>
      <p>
        Sign up
      </p>
    </div>
  </header>
  <section
    class="MainContainer"
  >
    <section
      class="generatorConfiguration"
    >
      <div
        class="dish-form"
      >
        <form
          id="checkboxes"
        >
          <div
            class="radio-label"
          >
            <label
              for="breakfast"
            >
              Breakfast
            </label>
            <input
              checked=""
              class="checkbox"
              id="breakfast"
              name="breakfast"
              type="radio"
              value="breakfast"
            />
          </div>
          <div
            class="radio-label"
          >
            <label
              for="lunch"
            >
              Lunch
            </label>
            <input
              class="checkbox"
              id="lunch"
              name="lunch"
              type="radio"
              value="lunch"
            />
          </div>
          <div
            class="radio-label"
          >
            <label
              for="dinner"
            >
              Dinner
            </label>
            <input
              class="checkbox"
              id="dinner"
              name="dinner"
              type="radio"
              value="dinner"
            />
          </div>
          <div
            class="radio-label"
          >
            <label
              for="dessert"
            >
              Dessert
            </label>
            <input
              class="checkbox"
              id="dessert"
              name="dessert"
              type="radio"
              value="dessert"
            />
          </div>
        </form>
      </div>
      <div
        class="dropdown-parent"
      >
        <label
          class="dropdown-label"
          for="Active Model:"
        >
          Active Model:
        </label>
        <select
          class="dropdown-select"
        >
          <option
            class="dropdown-option"
            value="mistral:7b"
          >
            mistral:7b
          </option>
          <option
            class="dropdown-option"
            value="text-davinci-003"
          >
            text-davinci-003
          </option>
          <option
            class="dropdown-option"
            value="llama2"
          >
            llama2
          </option>
        </select>
      </div>
    </section>
    <form
      class="IngredientForm"
    >
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1h51icj-MuiAutocomplete-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-l4u8b9-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id="autocompleteInput"
              placeholder="Add Ingredients..."
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <input
        disabled=""
        id="submit-button"
        type="submit"
        value="Generate Recipe"
      />
    </form>
    <br />
    <div
      class="containerTitle"
    >
      <h1
        class="active"
      >
        <em>
          Recipes
        </em>
      </h1>
      <h1
        class="inactive"
      >
        <em>
          Favorites
        </em>
      </h1>
    </div>
    <div
      class="RecipeContainer"
    >
      <div
        class="recipeDisplay"
      />
    </div>
  </section>
</div>
`;
